src/prometheus_alert_rules/heartbeat.rule

```diff
@@ -0,0 +1,7 @@
+# Based on https://awesome-prometheus-alerts.grep.to/rules.html#prometheus-self-monitoring-1
+alert: Watchdog
+expr: vector(1)
```

Comment:
**questions:** Do we know how an always-firing rule interacts with other alertmanager config? Should we document it here?

```
# ResolveTimeout is the default value used by alertmanager if the alert does
  # not include EndsAt, after this time passes it can declare the alert as resolved if it has not been updated.
  # This has no impact on alerts from Prometheus, as they always include EndsAt.
  [ resolve_timeout: [<duration>](https://prometheus.io/docs/alerting/latest/configuration/#duration) | default = 5m ]
```

```
# How long to initially wait to send a notification for a group
# of alerts. Allows to wait for an inhibiting alert to arrive or collect
# more initial alerts for the same group. (Usually ~0s to few minutes.)
[ group_wait: [<duration>](https://prometheus.io/docs/alerting/latest/configuration/#duration) | default = 30s ]
```

```
# How long to wait before sending a notification again if it has already
# been sent successfully for an alert. (Usually ~3h or more).
[ repeat_interval: [<duration>](https://prometheus.io/docs/alerting/latest/configuration/#duration) | default = 4h ]
```

Reply:
Can you be more specific about what you want to know? As far as I can tell, those three values don't work any differently with this alert rule.

Reply:
I mean the "always firing" aspect:
- What does "always" mean on the receiver side?
- Could it start firing a bit too late just because alertmanager config set a high `group_wait`?
- Would it stop firing against our intention because `resolve_timeout` is set?
- Could we miss an alert if cos-alerter was down for a bit and missed the `repeat_interval`?
- Any potential negative interaction between `repeat_interval` here and the time duration conf'd in cos-alerter?

Reply:
* I'm not sure I understand. Always means this alert always resolves to firing.
* Yes group_wait should be `0s`. I'm adding a note in the cos-alerter readme.
* From the docs you posted: "This has no impact on alerts from Prometheus, as they always include EndsAt."
* This would be okay as presumably we have Prometheus configured to alert when cos-alerter goes down.
* Yes it is already documented in cos-alerter that `down_interval` (configured in cos-alerter) should be longer than `repeat_interval`

Reply:
I have no idea but I don't think we specify that in any alert rules right?

Reply:
[This issue](https://github.com/prometheus/prometheus/issues/5277) seems to state that it is long enough that it won't be an issue.